core/src/main/java/io/grpc/util/RoundRobinLoadBalancerFactory.java

```diff
@@ -451,9 +451,36 @@ private Subchannel nextSubchannel() {
       return list;
     }
 
-    @VisibleForTesting
-    Status getStatus() {
-      return status;
+    @Override
+    boolean isEquivalentTo(RoundRobinPicker picker) {
+      if (!(picker instanceof ReadyPicker)) {
+        return false;
+      }
+      ReadyPicker other = (ReadyPicker) picker;
+      // the lists cannot contain duplicate subchannels
+      return other == this || (stickinessState == other.stickinessState
+          && list.size() == other.list.size()
+          && new HashSet<Subchannel>(list).containsAll(other.list));
+    }
+  }
+
+  @VisibleForTesting
+  static final class EmptyPicker extends RoundRobinPicker {
+    @Nullable
+    private final Status status;
+
+    EmptyPicker(@Nullable Status status) {
+      this.status = status;
```

Comment:
Making it non-Nullable, and using OK to indicate `PickResult.withNoResult()` may simplify the code a little bit.